LLM based Data Mapping

amit-sharma-d4l opened the pull request:
Please cross check this list if additions / modifications needs to be done on top of your core changes and tick them off. Reviewer can as well glance through and help the developer if something is missed out.

- [ ] Automated Tests (Jasmine integration tests, Unit tests, and/or Performance tests)
- [ ] Updated Manual tests / Demo Config
- [ ] Documentation (Application guide, Admin guide, Markdown, Readme and/or Wiki)
- [ ] Verified that local development environment is working with latest changes (integrated with latest `develop` branch)
- [ ] following best practices in [code review doc](../code_review.md)

amit-sharma-d4l committed:
data mapping to OMOP CDM - using LLM. Work in progress commit.

Signed-off-by: amit-sharma-d4l <amit.sharma@data4life-asia.care>

suwarnoong committed:
1498 show info for unmapped flow run result (#57)

suwarnoong committed:
1854 resize portal header to smaller (#60)

KangJingA committed:
[KangJingA/integrate 1901] merge data mapping flow to white rabbit (#65)

* create data mapping tasks

* update white rabbit flows with data mapping

* update ui

* remove unneeded endpoints

* remove data mapping flow

* fix link table button disabled after clicking back

* move task

* fix button

KangJingA committed:
[KangJingA/integrate 1901] fix failing mapping build (#66)

* create data mapping tasks

* update white rabbit flows with data mapping

* update ui

* remove unneeded endpoints

* remove data mapping flow

* fix link table button disabled after clicking back

* move task

* fix button

* remove unused components

mwaiyee committed:
[Adhoc] Fix GHA install d2e (#59)

* add if condition and parameterize version

amit-sharma-d4l committed:
Data mapping using LLM - final commit.
Description :
1. exposes /data-mapping route with http post
2. takes input in json format - same as the existing data mapping ui
3. returns the json format with mapping to OMOP tables - json to be parsed by the UI and draw arrows.

Signed-off-by: amit-sharma-d4l <amit.sharma@data4life-asia.care>

khairul-syazwan committed:
[analyze-1817] Add import ATLAS cohort JSON (#42)

* Add upload icon

* Add upload button to Bookmarks

* Add missing dialogWidth prop for MessageBox component

* Add new message box for JSON upload

* Add validator for Atlas Json import

* Refactor ImportAtlasCohortDefinitionDialog, minor css refactoring

* Remove load bookmarks query when closing import dialog

* Remove whitespace in css file

* Refactor to use Composition API

* Use script setup, add compatConfig

KangJingA commented on functions/data-mapping/src/data-mapping/services.ts:
```diff
@@ -0,0 +1,57 @@
+    return [mappedData, "200"];
+  } 
+  catch (error) {
+    return ["Error! invoking the LLM", "500"];
```
throw error instead of returning status code

KangJingA commented on functions/data-mapping/src/data-mapping/services.ts:
```diff
@@ -0,0 +1,57 @@
+  try {
+    const response = await model.invoke(messages);
+    const mappedData = response.content;
+    return [mappedData, "200"];
```
only return mappedData

KangJingA commented on functions/data-mapping/src/data-mapping/services.ts:
```diff
@@ -0,0 +1,57 @@
+  const [model, status] = await getModels(uiCode.model);
+  
+  // convert the received JSON data to required format
+  type DATA = {
```
move out type into another file, can give a more explicit name.

KangJingA commented on docker-compose-local.yml:
```diff
@@ -1,9 +1,8 @@
 services:
   trex:
-    command: npm run startdev
```
are these supposed to be removed?

KangJingA commented on docker-compose-local.yml:
```diff
@@ -35,6 +34,7 @@ services:
           "concept-mapping": false,
           "terminology-svc": false,
           "code-suggestion": false,
+          "data-mapping": true,
```
change to false

KangJingA commented on docker-compose.yml:
```diff
@@ -2,7 +2,6 @@ name: ${PROJECT_NAME:-d2e}
 networks:
   alp:
   data:
-  enterprise-gateway:
```
should not be removed

KangJingA commented on functions/data-mapping/src/data-mapping/routes.ts:
```diff
@@ -0,0 +1,31 @@
+
+  private registerRoutes() {
+    this.router.post("/", async (req: Request, res: Response) => {
+      req.body.model = AI_MODEL;
```
since only `getModels` function uses this value, can the service import this value instead of being passed through the request?

csafreen committed:
FHIR Server: Use portal dataset token study code to create project (#78)

csafreen committed:
Add scope for endpoint to work with client credential flow (#83)

csafreen committed:
Remove positional arguement (#86)

suwarnoong committed:
1851 add prefect blocks ui (#72)

jerome-ng committed:
Jerome ng/1879 improve search performance for duckdb (#77)

* Optimise search scoring

* Remove lower scoring criteria to improve speed

* improve filter options speed

* Remove filter option count

alicia-koh committed:
[data-1831-1832] Update flows documentation (#26)

* Update flows README
* Add scripts to generate package json, openapi spec
* Add static files
* Update dc local example mount for flows
* Update flows
* Regenerate package.json for all flows
* Switch images to ohdsi org
* Update white rabbit package.json
* Change image prefix

suwarnoong committed:
1906 fix button hover & loading color (#84)

suwarnoong committed:
1852 add prefect variables ui (#80)

amit-sharma-d4l committed:
Code restructuring - work in progress commit.

Signed-off-by: amit-sharma-d4l <amit.sharma@data4life-asia.care>

satish-a0 committed:
Optimize analytics config calls (#74)

* switch to pa config svc

* Embed paconfig metadata in request

* Inject pa config ID in request obj of analytics-svc

satish-a0 committed:
Optimize terminology-svc (#22)

* Fix properties for hana

* Fix including base concept ids

* Add async await

* Add keepalive terminology

* Filter for empty conceptsets

* Prepopulate dataset db details

* Promise all

* Parallelize

jerome-ng committed:
Jerome ng/1841 reduce min width filtercards section (#81)

* Remove icons in bookmark buttons

* Update spacing

* reduce width of left side

amit-sharma-d4l committed:
Added exception handling - work in progress commit

Signed-off-by: amit-sharma-d4l <amit.sharma@data4life-asia.care>

amit-sharma-d4l committed:
added exception handling and fixed naming issues within files

Signed-off-by: amit-sharma-d4l <amit.sharma@data4life-asia.care>

KangJingA committed:
[KangJingA/integrate 1853] fix prefect code cursor (#91)

* format css

* fix font size

suwarnoong committed:
1854 fix logo center (#87)

amit-sharma-d4l committed:
Formatting using prettier extension

Signed-off-by: amit-sharma-d4l <amit.sharma@data4life-asia.care>

suwarnoong committed:
1910 remove slot on UI, default slot name if pub is defined (#93)

SantanM committed:
[internal-1828] upgrade Strategus version, remove keyring creation and its references (#38)

* upgrade Strategus version, remove keyring creation and its references

* upgrade HADES, cohortgen and cohort incidence

brandantck committed:
[Analyze 1675] Add is excluded functionality for concept sets (#58)

* add isExcluded to frontend

* add isExcluded logic for backend

* fix invalid reason where clause

amit-sharma-d4l committed:
1. Removed unnecessary permissions
2. Declared a type for incoming JSON object from UI - previously it was a string, now it would be JSON object
3. Return from the API is also JSON object now - previously it was string.

Signed-off-by: amit-sharma-d4l <amit.sharma@data4life-asia.care>

mwaiyee committed:
[Devops-1869-1877] Add checkflow to cli (#79)

* add .mjs script to check flow status for max duration of 15 mins
* add checkflow and jupyter profile in cli

suwarnoong committed:
[Integrate-1908] rename db reader, mark sql query as stable (#89)

* 1908 rename db reader, mark sql query as stable

* 1908 remove SQL query node

suwarnoong committed:
1851 fix blocks icon background color (#90)

csafreen committed:
Reduce flow base image size (#75)

* Refactor to flow base image size

* Move plugins out of base flow

* Update docker file

* Update image name to ohdsi

* FIx bugs

* Fix DQD and data characterisation

* Fix strategus

* Fix path in dockerfile

* Fix bug

* Add new plugins in build

* Fix bugs

* Bug fix

* Set python library setuo

* Fix strategus image name

* Refactor to flow base image size

* Move plugins out of base flow

* Fix DQD and data characterisation

* Fix bugs

* Fix merge issue

* Add latest changes to strategus

* Add latest Strategus changes

* Add Achillies

* Remove duplicate

* Merge conflicts

* Merge conflict

* Change package.json

---------

Signed-off-by: Afreen <afreen.sikandara@data4life-asia.care>

mwaiyee committed:
[Devops-1930] GHA npm run checkflow (#100)

* change to process.exit in scripts upon failure in gha
* add checkflow step in GHA npm-install-run.yml

suwarnoong committed:
[Integrate-1907] use suggested mapping (#102)

* 1907 use suggested mapping

* 1907 mapping suggestion feature flag

Zhimin-arya committed:
[internal 1873] Concept hybrid_search (#95)

* concept hybrid_search

* getHybridConfig

* modify the search_embedding_plugin

amit-sharma-d4l committed:
Merge branch 'develop' into amit-sharma-d4l/data-mapping

Signed-off-by: amit-sharma-d4l <amit.sharma@data4life-asia.care>

amit-sharma-d4l committed:
[amit] changed some whitespace related merge conflicts

amit-sharma-d4l committed:
[amit] resolved merge conflicats - due to GPG key issue for signing commits

amit-sharma-d4l committed:
Merge branch 'develop' into amit-sharma-d4l/data-mapping

amit-sharma-d4l committed:
added post method permissions - left during merge conflict
